File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: list[int], n: int, maxSum: int) -> int:
        banned_set = set(banned)  
        count = 0
        
        for i in range(1, n + 1):
            if i not in banned_set and i <= maxSum:
                count += 1
                maxSum -= i
        
        return count

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        Set<Integer> bannedSet = new HashSet<>();
        for (int num : banned) {
            bannedSet.add(num);  // Add banned numbers to the set
        }
        
        int count = 0;
        for (int i = 1; i <= n; i++) {
            if (!bannedSet.contains(i) && i <= maxSum) {
                count++;
                maxSum -= i;
            }
        }
        
        return count;
    }
}
